shell/platform/embedder/fixtures/main.dart

```diff
@@ -879,3 +879,22 @@ void nativeArgumentsCallback(List<String> args) native 'NativeArgumentsCallback'
 void dart_entrypoint_args(List<String> args) {
   nativeArgumentsCallback(args);
 }
+
+void snapshotDimensionsCallback(int width, int height) native 'SnapshotDimensionsCallback';
+
+@pragma('vm:entry-point')
+void snapshot_large_scene() {
+  ReceivePort port = ReceivePort();
+  port.listen((dynamic max_size) async {
```

Comment:
We hadn't though of arranging the test this way before :) It totally works fine.

Another way of doing this would have been to call the default dart entrypoint which does nothing. Then, when the engine instance is running, Dart_Invoke a method that provides the `max_size`. This, instead of calling main, waiting in native code for main to be called, setting up a port and listener and then using that to communicate the max size. Each test case doesn't need a new "main". I just looked at the existing embedder_unittests and didn't see another test that uses this technique so its understandable that you missed it. It's tonic::DartInvoke or just the raw Dart_Invoke. Anyway. just an FYI as this is fine too.

Reply:
Oh, that's much simpler, thanks. I toured the runtime a bit and changed it up to use `DartInvoke`. However, I didn't quite figure out how to properly convert futures being returned from the Dart side, so it still has a callback going on. ¯\_(ツ)_/¯
I didn't come across any, but if you happen to know about any spots in the engine where a Dart future is getting converted/cast into something workable on the cpp side, I'd be interested in checking it out.

Either way, this turned out to be an excellent little issue for learning a number of things about skia + dart and the lifecycle of things. Thanks for the guidance. :)